backend/dataall/modules/maintenance/services/maintenance_service.py

```diff
@@ -0,0 +1,153 @@
+"""
+A service layer for maintenance activity
+Defines functions and business logic to be performed for maintenance window
+"""
+
+import logging
+import os
+
+from dataall.base.aws.event_bridge import EventBridge
+from dataall.base.aws.parameter_store import ParameterStoreManager
+from dataall.core.permissions.services.tenant_policy_service import TenantPolicyValidationService
+from dataall.modules.maintenance.api.enums import MaintenanceStatus
+from dataall.modules.maintenance.db.maintenance_repository import MaintenanceRepository
+from dataall.core.stacks.aws.ecs import Ecs
+
+logger = logging.getLogger(__name__)
+
+
+class MaintenanceService:
+    @staticmethod
+    def start_maintenance_window(engine, mode: str = None, groups=None):
+        """
+        Start maintenance window by performing following actions
+            1. Perform validation to check if the user belongs to the DAAdministrators group
+            2. Put the maintenance window status to PENDING and update the maintenance mode
+            3. Get all the ECS Scheduled tasks and disable the schedule for them
+        @param engine: db engine
+        @param mode: mode to set for maintenance window
+        @param groups: user groups from context.groups
+        @return: returns True if successful or False
+        """
+        # Check from the context if the groups contains the DAAAdminstrators group
+        if groups is None:
+            groups = []
+        if not TenantPolicyValidationService.is_tenant_admin(groups):
+            raise Exception('Only data.all admin group members can start maintenance window')
+
+        logger.info('Putting data.all into maintenance')
+        try:
+            with engine.scoped_session() as session:
```

Comment:
See comment in resolvers first. This should change to:

`with get_context().db_engine.scoped_session() as session`